gen-apidocs/generators/html.go

```diff
@@ -532,62 +557,75 @@ func (h *HTMLWriter) generateNavContent() string {
 	return nav
 }
 
-func (h *HTMLWriter) generateNavJS() {
-	// generate NavData
-	var tmp string
-	flatToc := []string{}
-	os.MkdirAll(api.BuildDir, os.ModePerm)
+type javascriptNavdata struct {
+	TOC     []javascriptTOCItem
+	FlatTOC []string
+}
+
+type javascriptTOCItem struct {
+	Section     string              `json:"section"`
+	Subsections []javascriptTOCItem `json:"subsections"`
+}
+
+func convertTOCItem(navdata *javascriptNavdata, item *TOCItem) javascriptTOCItem {
+	navdata.FlatTOC = append(navdata.FlatTOC, item.Link)
+
+	jsItem := javascriptTOCItem{
+		Section:     item.Link,
+		Subsections: []javascriptTOCItem{},
+	}
+
+	for _, subitem := range item.SubSections {
+		jsItem.Subsections = append(jsItem.Subsections, convertTOCItem(navdata, subitem))
+	}
+
+	return jsItem
+}
+
+func (h *HTMLWriter) generateNavJS() error {
```

Comment:
This whole function was replaced with a recursive collector (`convertTOCItem`) and simply using the regular Go JSON encoder. However there are 3 concerns:

1. I left out the `if strings.Contains(sec.Link, "strong")` logic, yet my code seems to produce the exact same data as the old code. Not quite sure why and what exactly the old code was attempting to do.
2.  The order of items in both the TOC and flatToc are now different, I presume the order is not important?
3. There seem to be a lot of duplication in the generated TOC. Whole top-level sections with all their subsections are duplicated in the generated code, both in the old and new implementation. Is this intended?